### DISCUSSION TITLE
Building iOS applications on Concourse

### DISCUSSION THREAD

**Discussion by pivotal-gordon-krull** [concourse/concourse#7150] [Help & Support]
Hello, are there any resources for building iOS apps using Concourse? I recall previously (years ago) using a local iMac as a worker to achieve this, but not sure the intended approach now that docker compose is the default method of spinning up a Concourse instance.

**Comment by aoldershaw** [comment]
The approach is still to run the `concourse worker` process on a Mac directly, and point it at a running Concourse instance (which could be deployed through `docker-compose` or however else).

Here's how we provision our Darwin worker: https://github.com/concourse/infrastructure/blob/master/terraform/dependencies/darwin-worker/scripts/startup.sh.tmpl

...and here's the `concourse` command that gets run: https://github.com/concourse/infrastructure/blob/master/terraform/dependencies/darwin-worker/scripts/concourse.sh.tmpl

As long as the provided `--tsa-worker-private-key` is trusted by the remote Concourse instance (by adding the corresponding public key to the configured `CONCOURSE_TSA_AUTHORIZED_KEYS` file), it'll be able to run `task`s (provided the task's `platform` is set to `darwin`)

**Reply by pivotal-gordon-krull** [reply]
Thank you so much @aoldershaw! This looks insanely helpful :)
